geps/gep-1016.yaml

```diff
@@ -0,0 +1,9 @@
+apiVersion: gateway.networking.k8s.io/v1alpha1
```

Comment:
Small procedural thing - I think I'd rather move to a directory per GEP so we end up with something like `gep-1234/metadata.yaml` and `gep-1234/index.md`, also corresponding to the upstream k/enhancements repo. Our GEPs directory is already massive and this current change would double it.

Reply:
Sigh, that makes sense. I'll see what I can do there.

Reply:
Okay, done. Whew, this is now a much bigger PR.